charms/openfga-k8s/src/charm.py

```diff
@@ -184,20 +185,36 @@ def _on_update_status(self, _):
         self._ready()
 
     @requires_state
-    def _create_token_secret(self, event):
-        if not self._state.token_secret_id:
+    def _create_token(self, event):
+        if JujuVersion.from_environ().has_secrets:
+            if not self._state.token_secret_id:
+                token = secrets.token_urlsafe(32)
+                content = {"token": token}
+                secret = self.app.add_secret(content)
+                self._state.token_secret_id = secret.id
+                logger.info("created token secret {}".format(secret.id))
```

Comment:
Don't we need `self._state.token = token` after this line?

Reply:
no.. we just send the secret_id via relation data bags